cmake/pika_add_config_test.cmake

```diff
@@ -447,7 +449,7 @@ function(pika_check_for_cxx23_static_call_operator)
   pika_add_config_test(
     PIKA_WITH_CXX23_STATIC_CALL_OPERATOR
     SOURCE cmake/tests/cxx23_static_call_operator.cpp
-    FILE ${ARGN}
+    FILE ${ARGN} CXXFLAGS -pedantic-errors
```

Comment:
Do you know what the failure mode is if this flag isn't supported? Does the test fail or does configuration fail? I'm thinking e.g. about `nvc++` which may not support the flag, but if it does it's perhaps not such a big concern.

Reply:
Reverted in https://github.com/pika-org/pika/pull/1369/commits/39b21c0cf970d74a1e914c60701c2adabbf7db95.